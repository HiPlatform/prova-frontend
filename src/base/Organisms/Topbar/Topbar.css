.topbar {
  width: 100%;
  height: 50px;
  position: fixed;
  top: 0;
  display: flex;
  flex-direction: row;
  background-color: #545454;
  margin-left: -8px;
  padding-left: 8px;
  border-radius: 3px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.2);
  z-index: 1;
  align-items: center;
  justify-content: center; }
  .topbar__back {
    font-size: 14px;
    color: #fff;
    text-shadow: #eff0f1;
    padding: 0px 10px; }
  .topbar__title {
    font-size: 14px;
    padding-left: 10px;
    color: #fff;
    text-shadow: #eff0f1;
    flex-grow: 1; }
