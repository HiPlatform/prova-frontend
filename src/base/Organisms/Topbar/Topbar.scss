@import "../../../variables";

.topbar {
  width: 100%;
  height: 50px;
  position: fixed;
  top: 0;
  display: flex;
  flex-direction: row;
  background-color: $solidLightBackground;
  margin-left: -8px;
  padding-left: 8px;
  border-radius: 3px;
  box-shadow: $shadow;
  z-index: 1;
  align-items: center;
  justify-content: center;

  &__back, button {
    font-size: 14px;
    color: $color;
    text-shadow: $textShadow;
    padding: 0px 10px;
    background-color: transparent;
    border: none;
  }

  &__title {
    font-size: 14px;
    padding-left: 10px;
    color: $color;
    text-shadow: $textShadow;
    flex-grow: 1;
  }
}