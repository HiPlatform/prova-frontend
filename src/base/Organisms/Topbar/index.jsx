import React from 'react';
import PropTypes from 'prop-types';

import './Topbar.css';

const Topbar = ({ navigate }) => (
  <div className="topbar">
    <h1 className="topbar__back">
      <button onClick={() => navigate('home')}>
        <i className="fa fa-chevron-left" />
      </button>
    </h1>
    <h1 className="topbar__title">Music Tracker</h1>
  </div>
);

Topbar.propTypes = {
  navigate: PropTypes.func.isRequired,
};

export default Topbar;
