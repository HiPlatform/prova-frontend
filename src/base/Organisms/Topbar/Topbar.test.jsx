import React from 'react';
import { shallow } from 'enzyme';
import Topbar from './index';

describe('Topbar', () => {
  const navigate = jest.fn();
  const props = {
    navigate,
  };
  const component = shallow(<Topbar {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the navigate function when the "back" button is clicked', () => {
    component.find('.topbar__back > button').simulate('click');
    expect(navigate).toHaveBeenCalled();
  });
});
