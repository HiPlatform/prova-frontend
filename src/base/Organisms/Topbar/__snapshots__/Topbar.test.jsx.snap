// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topbar
    navigate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="topbar__back"
>
          <button
                    onClick={[Function]}
          >
                    <i
                              className="fa fa-chevron-left"
                    />
          </button>
</h1>,
        <h1
          className="topbar__title"
>
          Music Tracker
</h1>,
      ],
      "className": "topbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
>
            <i
                        className="fa fa-chevron-left"
            />
</button>,
          "className": "topbar__back",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-chevron-left"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-chevron-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Music Tracker",
          "className": "topbar__title",
        },
        "ref": null,
        "rendered": "Music Tracker",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="topbar__back"
>
            <button
                        onClick={[Function]}
            >
                        <i
                                    className="fa fa-chevron-left"
                        />
            </button>
</h1>,
          <h1
            className="topbar__title"
>
            Music Tracker
</h1>,
        ],
        "className": "topbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
>
              <i
                            className="fa fa-chevron-left"
              />
</button>,
            "className": "topbar__back",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-chevron-left"
/>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-chevron-left",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Music Tracker",
            "className": "topbar__title",
          },
          "ref": null,
          "rendered": "Music Tracker",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
